servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java

```diff
@@ -1221,4 +1221,73 @@ public void fetchEntriesByNamelessReference() throws BaseNessieClientServerExcep
             immutableEntry(a, Content.Type.ICEBERG_TABLE),
             immutableEntry(b, Content.Type.ICEBERG_VIEW));
   }
+
+  /**
+   * This test verifies that a merge with squash strategy correctly merges operations targeting the
+   * same key, and that the result commit does not contain duplicate keys or spurious operations.
+   */
+  @Test
+  public void commitMergeSquashSameKeys() throws Exception {
```

Comment:
I added a test for merge, as requested. I wanted to also add a test for transplant, but realized that the API does not allow transplants with squash strategy. The version store layer does accept that though.

Reply:
Done.